account_invoice_shipping_address/stock.py

```diff
@@ -19,16 +19,23 @@
 #
 ##############################################################################
 
-from openerp.osv import orm
+from openerp import models
 
 
-class StockPicking(orm.Model):
+class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     def _get_invoice_vals(self, cr, uid, key, inv_type,
-                          journal_id, picking, context=None):
+                          journal_id, move, context=None):
         invoice_vals = super(StockPicking, self)._get_invoice_vals(
-            cr, uid, key, inv_type, journal_id, picking, context=context)
-        if picking and picking.partner_id:
-            invoice_vals['address_shipping_id'] = picking.partner_id.id
+            cr, uid, key, inv_type, journal_id, move, context=context)
+        if move and move.partner_id:
+            if 'delivery_address_id' in self.env['stock.picking']._fields:
+                invoice_vals['address_shipping_id'] = (
+                    move.picking_id.delivery_address_id and
```

Comment:
You can use directly `move.picking_id.delivery_address_id.id` without the fear of errors.